Assert.Throws ambiguity between `Func<object>, string` and `Action, AssertNonStrictThrowsInterpolatedStringHandler` overloads

## Describe the bug

> error CS0121: The call is ambiguous between the following methods or properties: 'Assert.Throws<TException>(Func<object?>, string, params object[])' and 'Assert.Throws<TException>(Action, ref Assert.AssertNonStrictThrowsInterpolatedStringHandler<TException>)'

NOTE: This is a regression.

## Steps To Reproduce

```csharp
Assert.Throws<Exception>(() => Console.ReadLine(), $"Hello {0}");
```

## Expected behavior

It should compile

## Actual behavior

Ambiguous

## Additional context

Adding an extra overload for interpolated string for `Func<object?>` resolves this.

--- a/src/TestFramework/TestFramework/Assertions/Assert.ThrowsException.cs
+++ b/src/TestFramework/TestFramework/Assertions/Assert.ThrowsException.cs
@@ -31,6 +31,11 @@ public sealed partial class Assert
             }
         }
 
+        public AssertNonStrictThrowsInterpolatedStringHandler(int literalLength, int formattedCount, Func<object?> action, out bool shouldAppend)
+            : this(literalLength, formattedCount, (Action)(() => _ = action()), out shouldAppend)
+        {
+        }
+
         internal TException ComputeAssertion()
         {
             if (_state.FailAction is not null)
@@ -214,6 +219,13 @@ public sealed partial class Assert
     /// <inheritdoc cref="Throws{TException}(Action, string, object[])" />
 #pragma warning disable IDE0060 // Remove unused parameter - https://github.com/dotnet/roslyn/issues/76578
     public static TException Throws<TException>(Action action, [InterpolatedStringHandlerArgument(nameof(action))] ref AssertNonStrictThrowsInterpolatedStringHandler<TException> message)
+#pragma warning restore IDE0060 // Remove unused parameter
+        where TException : Exception
+        => message.ComputeAssertion();
+
+    /// <inheritdoc cref="Throws{TException}(Action, string, object[])" />
+#pragma warning disable IDE0060 // Remove unused parameter - https://github.com/dotnet/roslyn/issues/76578
+    public static TException Throws<TException>(Func<object?> action, [InterpolatedStringHandlerArgument(nameof(action))] ref AssertNonStrictThrowsInterpolatedStringHandler<TException> message)
 #pragma warning restore IDE0060 // Remove unused parameter
         where TException : Exception
         => message.ComputeAssertion();

--- a/src/TestFramework/TestFramework/PublicAPI/PublicAPI.Unshipped.txt
+++ b/src/TestFramework/TestFramework/PublicAPI/PublicAPI.Unshipped.txt
@@ -1,2 +1,4 @@
 #nullable enable
+Microsoft.VisualStudio.TestTools.UnitTesting.Assert.AssertNonStrictThrowsInterpolatedStringHandler<TException>.AssertNonStrictThrowsInterpolatedStringHandler(int literalLength, int formattedCount, System.Func<object?>! action, out bool shouldAppend) -> void
 Microsoft.VisualStudio.TestTools.UnitTesting.ConditionBaseAttribute.Mode.get -> Microsoft.VisualStudio.TestTools.UnitTesting.ConditionMode
+static Microsoft.VisualStudio.TestTools.UnitTesting.Assert.Throws<TException>(System.Func<object?>! action, ref Microsoft.VisualStudio.TestTools.UnitTesting.Assert.AssertNonStrictThrowsInterpolatedStringHandler<TException!> message) -> TException!

--- a/test/UnitTests/TestFramework.UnitTests/Assertions/AssertTests.ThrowsExceptionTests.cs
+++ b/test/UnitTests/TestFramework.UnitTests/Assertions/AssertTests.ThrowsExceptionTests.cs
@@ -202,6 +202,18 @@ public partial class AssertTests
         Verify(ex is AssertFailedException);
     }
 
+    public void Throws_WithInterpolation()
+    {
+        static string GetString() => throw new Exception();
+
+#pragma warning disable IDE0200 // Remove unnecessary lambda expression - intentionally testing overload resolution for this case.
+        Exception ex = Assert.Throws<Exception>(() => GetString(), $"Hello {GetString()}");
+#pragma warning restore IDE0200 // Remove unnecessary lambda expression
+        Exception ex2 = Assert.Throws<Exception>(GetString, $"Hello {GetString()}");
+        Verify(ex is not null);
+        Verify(ex2 is not null);
+    }
+
     public void ThrowsExactly_WhenExceptionIsDerivedFromExpectedType_ShouldThrow()
     {
         static void Action() => Assert.ThrowsExactly<ArgumentException>(() => throw new ArgumentNullException());

